evm/src/fuzz/mod.rs

```diff
@@ -196,9 +196,11 @@ impl<'a> FuzzedExecutor<'a> {
                 let reason = reason.to_string();
                 result.reason = if reason.is_empty() { None } else { Some(reason) };
 
-                let args = func
-                    .decode_input(&calldata.as_ref()[4..])
-                    .map_err(|_| FuzzError::FailedDecodeInput)?;
+                let args = if calldata.len() < 4 || func.inputs.is_empty() {
+                    vec![]
+                } else {
+                    func.decode_input(&calldata[4..]).map_err(|_| FuzzError::FailedDecodeInput)?
+                };
```

Comment:
what was the error here? empty calldata?

Reply:
it's related to the linked issue I'm still trying to figure it out

Reply:
fyi I usually track these things down by annotating everything with #[track_caller] to find where this originates.

I've seen similar panics, and it likely happens inside the `revm::EvmImpl` when working on JournaledState because there are unchecked operations, especially on revert handling.